src/components/ArticleCard.jsx

```diff
@@ -34,16 +34,52 @@ function Comments({ articleId }) {
 
 
 function ArticleCard() {
-  const { id } = useParams(); 
+  const { id } = useParams();
   const [article, setArticle] = useState(null);
+  const [voting, setVoting] = useState(false);
+  const [voteError, setVoteError] = useState(null);
 
   useEffect(() => {
-   
     fetch(`https://louis-api.onrender.com/api/articles/${id}`)
       .then((res) => res.json())
       .then((data) => setArticle(data.article));
   }, [id]);
 
+  const handleVote = (voteType) => {
+    if (voting) {
+      return;
+    }
+
+    setVoting(true);
+    setVoteError(null);
+
+    const voteValue = voteType === "upvote" ? 1 : -1; 
+
+    fetch(`https://louis-api.onrender.com/api/articles/${id}`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ inc_votes: voteValue }), 
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error("Failed to update votes. Please try again.");
```

Comment:
This is a great error message for a user, however the error message they are shown on the site is actually _"Failed to Fetch"_ which is a little less user friendly to an average joe, it might be worth seeing why that is happening so you can show your message instead